packages/components/src/styles/components/stepper/list.scss

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// STEPPER > LIST
+//
+
+$hds-stepper-list-step-padding-bottom: 16px;
+
+.hds-stepper-list,
+.hds-stepper-list__step {
+  margin: 0;
+  padding: 0;
+  list-style: none;
+}
+
+// SUB COMPONENT: STEPPER > LIST > STEP
+
+.hds-stepper-list__step {
+  display: grid;
+  grid-column-gap: 8px;
+  grid-template-columns: auto 1fr;
+
+  &:not(:last-of-type) {
+    padding-bottom: $hds-stepper-list-step-padding-bottom;
+
+    .hds-stepper-list__step-progress {
+      &::before,
+      &::after {
+        position: absolute;
+        top: 1px;
+        left: 11px;
+        z-index: -1;
+        width: 2px;
+        content: "";
+      }
+
+      &::before {
+        height: calc(100% + $hds-stepper-list-step-padding-bottom);
+        background: linear-gradient(0deg, var(--token-color-palette-alpha-200) 0%, var(--token-color-palette-alpha-200) 100%), var(--token-color-palette-neutral-100);
+      }
+
+      &::after {
+        background-color: var(--token-color-palette-neutral-700);
+      }
+    }
+  }
+}
+
+.hds-stepper-list__step-progress {
+  position: relative;
+  z-index: 0;
```

Comment:
It may be more explicit to use `isolation: isolate;` instead of `z-index: 0;`.

https://developer.mozilla.org/en-US/docs/Web/CSS/isolation